src/common/hooks/useDefaultProvider.ts

```diff
@@ -0,0 +1,13 @@
+import { useMemo } from "react";
+import { ethers, providers } from "ethers";
+import { NETWORK_NAME } from "../../config";
+
+export const useDefaultProvider = (): { provider: providers.Provider } => {
+  const provider = ethers.getDefaultProvider(NETWORK_NAME);
+  return useMemo(
+    () => ({
+      provider
+    }),
+    [] // eslint-disable-line react-hooks/exhaustive-deps
+  );
```

Comment:
```suggestion
  return useMemo(
    () => ({
      provider: ethers.getDefaultProvider(NETWORK_NAME);
    }),
    [NETWORK_NAME, ethers]
  );
```

Reply:
Yep good idea, this doesn't remove the need to remove exhautive-deps though. It requires `provider` to be in it which cannot be done because it will return a new instance each time, causing a infinity loop.

I'll add these two in and leave in the eslint-disable

Reply:
if you change line 9 to `provider: ethers.getDefaultProvider(NETWORK_NAME)`, there shouldnt be any problem :thinking: infinity loop still happen ?

Reply:
yep this fixes it!